src/tools/queryDatasource/validators/validateFilterValues.test.ts

```diff
@@ -0,0 +1,439 @@
+import { Err, Ok } from 'ts-results-es';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+import VizqlDataServiceMethods from '../../../sdks/tableau/methods/vizqlDataServiceMethods.js';
+import { Query } from '../queryDatasourceValidator.js';
+import { validateFilterValues } from './validateFilterValues.js';
+
+// Mock the VizqlDataServiceMethods
+const mockVizqlDataServiceMethods = {
+  queryDatasource: vi.fn(),
+} as unknown as VizqlDataServiceMethods;
+
+const mockDatasource = {
+  datasourceLuid: 'test-datasource-luid',
+};
+
+describe('validateFilterValues', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should return Ok when no filters are present', async () => {
+    const query: Query = {
+      fields: [{ fieldCaption: 'Sales' }],
+    };
+
+    const result = await validateFilterValues(query, mockVizqlDataServiceMethods, mockDatasource);
+
+    expect(result.isOk()).toBe(true);
+  });
+
+  it('should return Ok when no SET or MATCH filters are present', async () => {
+    const query: Query = {
+      fields: [{ fieldCaption: 'Sales' }],
+      filters: [
+        {
+          field: { fieldCaption: 'Sales' },
+          filterType: 'QUANTITATIVE_NUMERICAL',
+          quantitativeFilterType: 'MIN',
+          min: 1000,
+        },
+      ],
+    };
+
+    const result = await validateFilterValues(query, mockVizqlDataServiceMethods, mockDatasource);
+
+    expect(result.isOk()).toBe(true);
+  });
+
+  it('should validate SET filter with valid values', async () => {
+    const query: Query = {
+      fields: [{ fieldCaption: 'Sales' }],
+      filters: [
+        {
+          field: { fieldCaption: 'Region' },
+          filterType: 'SET',
+          values: ['East', 'West'],
+        },
+      ],
+    };
+
+    // Mock successful query returning existing values
+    (mockVizqlDataServiceMethods.queryDatasource as any).mockResolvedValue(
```

Comment:
Let's try to avoid using `as any`. How about a better type like `as unknown as ReturnType<typeof vi.fn>`?